doc/rationale.md

```diff
@@ -0,0 +1,31 @@
+# Ignition rationale
+
+Ignition is a distribution-agnostic provisioning utility. When a machine boots for the first time, Ignition will manipulate it in various ways to set up the machine for operations.
+
+There are several principles that have guided Ignition's design. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool, not a configuration management tool. Ignition encourages immutable infrastructure, wherein if a user wishes to modify a machine, they should discard the old node and re-provision the machine. This keeps the user's machines in a well known state with relatively simple tooling.
```

Comment:
... immutable infrastructure, in which machine modification requires that users discard the old node and re-provision the machine. This keeps ...